ci/jenkins/utils.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    echo "====== Checking installed Golang version ======"
+    installed_version=$(/usr/local/go/bin/go version | grep -oP "go\d+\.\d+(\.\d+)?")
+    antrea_golang_version="go$(go mod edit -json | jq -r .Go)"
+    if [[ "$installed_version" =~ "$antrea_golang_version" ]]; then
+        echo "====== Golang version on the testbed is $installed_version ======"
+        echo "====== No need to upgrade ======"
```

Comment:
If `installed_version` is `go1.21.0` and `antrea_golang_version` is `go1.21`, why don't we upgrade to the latest patch version?

Reply:
I'm not sure if we need to upgrade the testbed golang every time there's a new golang release. The golang version in `go.mod` will not contain the patch number, if we want to check the patch number we need to query `go.dev` every time the test job is triggered.

Reply:
Code updated. Now the golang version on the testbed will always be the latest patch version.